Assets/Scripts/Controllers/UtilitySpriteController.cs

```diff
@@ -0,0 +1,173 @@
+#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System.Collections.Generic;
+using UnityEngine;
+
+public class UtilitySpriteController : BaseSpriteController<Utility>
+{
+    private Dictionary<Utility, GameObject> powerStatusGameObjectMap;
+
+    // Use this for initialization
+    public UtilitySpriteController(World world) : base(world, "Utility")
+    {
+        // Register our callback so that our GameObject gets updated whenever
+        // the tile's type changes.
+        world.OnUtilityCreated += OnCreated;
+
+        // Go through any EXISTING utility (i.e. from a save that was loaded OnEnable) and call the OnCreated event manually.
+        foreach (Utility util in world.utilities)
+        {
+            OnCreated(util);
+        }
+    }
+
+    public override void RemoveAll()
+    {
+        world.OnUtilityCreated -= OnCreated;
+
+        foreach (Utility util in world.utilities)
+        {
+            util.Changed -= OnChanged;
+            util.Removed -= OnRemoved;
+        }
+
+        foreach (Utility util in powerStatusGameObjectMap.Keys)
+        {
+            GameObject.Destroy(powerStatusGameObjectMap[util]);
+        }
+            
+        powerStatusGameObjectMap.Clear();
+        base.RemoveAll();
+    }
+
+    public Sprite GetSpriteForUtility(string objectType)
+    {
+        Utility proto = PrototypeManager.Utility.Get(objectType);
+        Sprite s = SpriteManager.current.GetSprite("Utility", objectType + (proto.LinksToNeighbour ? "_" : string.Empty));
```

Comment:
`return SpriteManager.current.GetSprite("Utility", objectType + (proto.LinksToNeighbour ? "_" : string.Empty));`

Than you would get rid of disgusting `s`